src/elements/progress-spinner/progress-spinner.md

```diff
@@ -0,0 +1,51 @@
+### Indeterminate
+
+Indicates infinite loading.
+
+```jsx
+<ProgressSpinner />
+```
+
+### Controlled
+
+Indicates the actual progress value.
+
+```jsx
+import { useEffect, useRef, useState } from 'react'
+import Button from '../button';
+
+const AnimatedProgress = () => {
+  const timeoutRef = useRef()
+  
+  const [value, setValue] = useState(0)
+  const maxValue = 100
+  
+  useEffect(() => {
+    if (value < maxValue) {
+      timeoutRef.current = setTimeout(() => {
+        setValue(value + 1)
+      }, 20)
+    }
+    
+    return () => clearTimeout(timeoutRef.current)
+  })
+
+  return (
+    <>
+      <p>
+        <ProgressSpinner
+          value={value}
+          max={maxValue}
+          style={{ marginRight: '0.5rem' }}
+        />
+        Loading...
+      </p>
+      <Button variant="contained" onClick={() => { setValue(0) }} disabled={value < maxValue}>
+        Restart
+      </Button>
```

Comment:
Is it intentional then loading animation starts from the quarter?

Reply:
It's a trick I do not really know how to deal with in the example. When you click 'restart', the value immediately changes to 0 and starts increasing by 1 every 20 milliseconds. In the same time, we have a transition for 250 milliseconds that starts immediately after the button click too.

The `transition` decreases the actual visual value by 1 every `2.5ms` when the `setTimeout` increases the value by 1 every `20ms`. Putting it together: `2.5 × (100 - x) = 20x` → They meet at the `x = 11` (11% of the path from the start).

Reply:
Although, it does not work as expected and it is actually after the quarter. I have no idea why

Reply:
Me neither to be honest. This SVG stuff is very new to me. 😕